docs/Kubernetes.md

```diff
@@ -0,0 +1,97 @@
+Table of Contents
+---
+
+- [Kubernetes usage](#kubernetes-usage)
+  - [Runtime watcher](#runtime-watcher)
+  - [Operation execution worker](#operation-execution-worker)
+- [Configuring cluster access](#configuring-cluster-access)
+  - [inCluster mode](#using-incluster-mode)
+  - [kubeconfig mode](#using-kubeconfig-mode)
+
+## Kubernetes usage
+Maestro uses kubernetes for orchestrating game room instances. It uses a unique **namespace** for each scheduler, and a unique **pod** for each game room instance.
+
+We use [client-go](https://github.com/kubernetes/client-go) for communicating with kubernetes. The [Runtime](internal/core/ports/runtime.go) port
+is the interface used for managing resources, you can find all of the features we are using for managing k8s resources in it.
+
+The diagram below shows how maestro components interact with kubernetes for managing resources.
+
+```mermaid
+flowchart BT
+classDef borderless stroke-width:0px
+classDef darkBlue fill:#00008B, color:#fff
+classDef brightBlue fill:#6082B6, color:#fff
+classDef gray fill:#62524F, color:#fff
+classDef gray2 fill:#4F625B, color:#fff
+
+
+subgraph maestroSystem[ ]
+    subgraph k8s[ ]
+        A3[Kubernetes]
+    end
+    class k8s,A3 brightBlue
+    class A3, borderless
+    
+    subgraph WORKER[ ]
+        A7[Operation Execution Worker<br/><br/>manage kubernetes resources by creating/deleting/updating pods abnd namespaces]
+    end
+    class WORKER,A7 brightBlue
+    class WORKER,A7 borderless
+    WORKER--Create namespace<br/>HTTPS-->k8s
+    WORKER--Delete namespace<br/>HTTPS-->k8s
+    WORKER--Create pod<br/>HTTPS-->k8s
+    WORKER--Delete pod<br/>HTTPS-->k8s
+    
+    subgraph RUNTIME_WATCHER[ ]
+        A8[Runtime Watcher <br/><br/> watch for change events in managed pods]
+    end
+    class RUNTIME_WATCHER,A8 brightBlue
+    class RUNTIME_WATCHER,A8 borderless
+    RUNTIME_WATCHER--List/Watch pods<br/>HTTPS-->k8s
+    
+
+    
+end
+
+
+click A3 "/csymapp/mermaid-c4-model/blob/master/AWAComponent.md" "AWA"
+
+```
+
+### Runtime watcher
+The runtime watcher component maintains a worker process for each scheduler that keeps watching and processing _change
+events_ in pods resources. For doing that, it uses a [pods informer](https://medium.com/codex/explore-client-go-informer-patterns-4415bb5f1fbd),
```

Comment:
What do you think about adding an official go pkg doc instead of a medium text? I think it's complicated to make references for a text written by someone we do not know and it's not related to maestro or the official pkg
```suggestion
events_ in pods resources. For doing that, it uses a [pods informer](https://pkg.go.dev/k8s.io/client-go/informers),
```